docs/framework/winforms/windows-forms-accessibility-improvements.md

```diff
@@ -0,0 +1,96 @@
+
+__Improved CheckedListBox Accessibility support__
+- Corrected CheckedListBox Bounds provided by accessibility properties for entries. Improved overall ListBox and CheckedListBox accessibility (corrected pproperty values and event model).
+
+__Improved ComboBox Accessibility support__
+- Updated the process of getting ComboBox items' accessibility objects to enable generating IDs for items instead of getting hash codes from items which may be unsafe in case of GetHashCode function is overriden.
+
+__Improved DataGridView Accessibility support__
+- Corrected DataGridView Bounds provided by accessibility properties for columns, rows, cells and corresponding headers, improved performance of bounding rectangle calculation. For now all accessibility bounds are represented correctly taking into account the bounds of entire control, its viewport and scrolls.
+- Corrected Value.IsReadOnly property value providing for accessible client applications. The property now shows correct IsReadOnly status for cells.
+- Fixed the issue with CellParsing event raising for the very first cell change. Cell value is now can be changed without any issues including the very first DataGridView control interaction.
+
+__Improved PropertyGrid Accessibility support__
+- Corrected PropertyGrid Bounds provided by accessibility properties for grid entries, improved performance of bounding rectangle calculation. For now all accessibility bounds are represented correctly taking into account the bounds of entire control, its viewport and scrolls.
+- Corrected accessible names and descriptions of subcontrols to not include control type names and to avoid double announcement for control type names.
+
+__Improved ToolStrip Accessibility support__
+- Improved navigation through ToolStrip items. Corrected ToolStrip and Menu shift-tab navigation (back-looping the menu items when shift-tab up-arrow is pressed - navigate to the bottom menu element)
+- Improved Menu accessible navigation, corrected menu accessible control types for submenus to make submenus of type 'Menu' instead of 'MenuItem'.
+
+__Improved PrintPreviewControl and PrintPreviewDialog Accessibility support__
+- Improved accessible navigation (including Narrator navigation) through menu items.
+- Improved High Contrast themes support and made the control element more contrasted.
+
+__StringCollectionEditor Accessibility support__
+- Added StringCollectionEditor with improved accessibility support.
+
+__UI automation support for DataGridView, PropertyGrid, ListBox, ComboBox, ToolStrip and other controls__
+Note: UI automation support is enabled for controls in runtime but is not used in design time.
+For an overview of UI automation, see the [UI Automation Overview](https://docs.microsoft.com/dotnet/framework/ui-automation/ui-automation-overview).
+
+ 
```

Comment:
Lines 65 through 70 can be deleted. Optionally, the remaining commented-out list can be deleted.

Reply:
Fixed, thanks.